services/web/server/src/simcore_service_webserver/diagnostics_monitoring.py

```diff
@@ -1,15 +1,15 @@
 """ Enables monitoring of some quantities needed for diagnostics
 
 """
+import concurrent.futures
 import logging
 import time
-from typing import Coroutine
+from typing import Callable, Coroutine
 
 import prometheus_client
 from aiohttp import web
 from prometheus_client import CONTENT_TYPE_LATEST, Counter, Gauge, Histogram
 from prometheus_client.registry import CollectorRegistry
```

Comment:
Sorry but why not replace it with the async version of the Prometheus client and be done with all the misery of blocking metrics?  See following link https://prometheus-async.readthedocs.io/en/stable/

Reply:
I knew about this library since sometime. It is indeed async out-of-the-box **but** at first sight it looks like a very different interfacing. The only bottleneck we have is generating the metrics response and the current fix makes it better. For the moment i would not invest more time into this since it is a side/optional feature. What do you think?

Reply:
Fine for now, if problems occur later we'll handle them